### PR TITLE
Update draft order validation

### PR DESCRIPTION
**Pull request by IKarbowiak**
- Extend `Order` type with `errors: [OrderError!]!` field. When queried, this field will run draft order validation.
- Update draft order validation - run all checks and return a list of errors
- Create tasks for deleting order lines by deleting products or variants

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2294 DraftOrder- canFinalize doesn't return any feedback what is incorrect in draft order](https://app.clickup.com/t/2549495/SALEOR-2294)

**Review by d-wysocki — APPROVED** [review]

File: saleor/graphql/core/types/common.py [hunk 1]
```diff
@@ -178,6 +178,11 @@ class OrderError(Error):
         description="List of product variants that are associated with the error",
         required=False,
     )
+    order_lines = graphene.List(
```
Comment (fowczarek) [inline saleor/graphql/core/types/common.py:181]: In this type, we have `order_line` and `order_lines`. IMO we should have one of these.

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -1089,3 +1096,13 @@ def wrap_voucher_with_channel_context(data):
     @traced_resolver
     def resolve_language_code_enum(root, _info, **_kwargs):
         return LanguageCodeEnum[str_to_enum(root.language_code)]
+
+    @staticmethod
+    @traced_resolver
+    def resolve_errors(root, _info, **_kwargs):
+        country = get_order_country(root)
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:1103]: This function could generate a lot of DB queries.

File: saleor/graphql/order/types.py [hunk 3]
```diff
@@ -1089,3 +1096,13 @@ def wrap_voucher_with_channel_context(data):
...
+        try:
+            validate_draft_order(root, country)
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:1105]: This function could generate a lot of DB queries.

**Review by fowczarek — CHANGES_REQUESTED** [review]

File: saleor/graphql/order/types.py (hunk 2 above)
Comment (IKarbowiak) [reply saleor/graphql/order/types.py:1100]: Will be optimized in separate PR.

File: saleor/graphql/order/types.py (hunk 3 above)
Comment (IKarbowiak) [reply saleor/graphql/order/types.py:1102]: Will be optimized in separate PR.

File: saleor/graphql/order/types.py (hunk 2 above)
Comment (IKarbowiak) [reply saleor/graphql/order/types.py:1100]: https://app.clickup.com/t/2549495/SALEOR-3355

File: saleor/graphql/order/types.py (hunk 3 above)
Comment (IKarbowiak) [reply saleor/graphql/order/types.py:1102]: https://app.clickup.com/t/2549495/SALEOR-3355

File: saleor/graphql/core/types/common.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/graphql/core/types/common.py]: I dropped `orderLine` and left only `orderLines`

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**